packages/pyolite-kernel/py/ipykernel/ipykernel/jsonutil.py

```diff
@@ -0,0 +1,165 @@
+"""Utilities to manipulate JSON objects."""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import math
+import numbers
+import re
+import types
+from binascii import b2a_base64
+from datetime import datetime
+
+# lite: cannot import this due to zmq
+# from jupyter_client._version import version_info as jupyter_client_version
+
+next_attr_name = "__next__"
+
+# -----------------------------------------------------------------------------
+# Globals and constants
+# -----------------------------------------------------------------------------
+
+# timestamp formats
+ISO8601 = "%Y-%m-%dT%H:%M:%S.%f"
+ISO8601_PAT = re.compile(
+    r"^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})(\.\d{1,6})?Z?([\+\-]\d{2}:?\d{2})?$"
+)
+
+# holy crap, strptime is not threadsafe.
+# Calling it once at import seems to help.
+datetime.strptime("1", "%d")
+
+# -----------------------------------------------------------------------------
+# Classes and functions
+# -----------------------------------------------------------------------------
+
+
+# constants for identifying png/jpeg data
+PNG = b"\x89PNG\r\n\x1a\n"
+# front of PNG base64-encoded
+PNG64 = b"iVBORw0KG"
+JPEG = b"\xff\xd8"
+# front of JPEG base64-encoded
+JPEG64 = b"/9"
+# constants for identifying gif data
+GIF_64 = b"R0lGODdh"
+GIF89_64 = b"R0lGODlh"
+# front of PDF base64-encoded
+PDF64 = b"JVBER"
+
+# lite: we do not know this
+# JUPYTER_CLIENT_MAJOR_VERSION = jupyter_client_version[0]
+
+
+def encode_images(format_dict):
```

Comment:
Is this actually needed if `format_dict` is returned as is?

Reply:
Maybe it can be removed to reduce the amount of vendored code.

Reply:
right: if it didn't need the patch, i was going to copy it from `sys.prefix` :stuck_out_tongue_closed_eyes:

Reply:
also in case we need to diff it against upstream at some point it's probably better to not touch it too much :+1: